tests/test_microgrid/test_timeout.py

```diff
@@ -0,0 +1,171 @@
+"""Benchmark for microgrid data.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import time
+from typing import Any, Iterator
+
+import grpc
+import pytest
+from frequenz.api.microgrid.microgrid_pb2 import (
+    ComponentData,
+    ComponentFilter,
+    ComponentIdParam,
+    ComponentList,
+    ConnectionFilter,
+    ConnectionList,
+    PowerLevelParam,
+    SetBoundsParam,
+)
+from google.protobuf.empty_pb2 import Empty  # pylint: disable=no-name-in-module
+from pytest_mock import MockerFixture
+
+from frequenz.sdk.microgrid.client import MicrogridGrpcClient
+
+from .mock_api import MockGrpcServer, MockMicrogridServicer
+
+
+async def test_components_timeout(mocker: MockerFixture) -> None:
+    """Test if the components() method properly raises AioRpcError"""
+    servicer = MockMicrogridServicer()
+
+    # pylint: disable=unused-argument
+    def mock_list_components(request: ComponentFilter, context: Any) -> ComponentList:
+        time.sleep(0.3)
```

Comment:
What are these small sleep/timeouts, `0.3`, `0.1` and `0.5`? Are we introducing new flaky tests? :worried: 

If we are not, then we should add comments about why the sleep/timeouts are there and why that value was picked.

Reply:
These are arbitrary values. The argument to `sleep` must be bigger than the argument passed to `timeout` so that the timeout is triggered. I just picked `0.3` and `0.1`, but could be something else. I can pull them out to some constants and add a comment there.

There's one test where I had to change the numbers, because the `<grpc-method>.write()` was used and I wasn't sure how to handle that case. Maybe someone knows, otherwise I will keep digging.

Reply:
Yes please, constant and comments clarifying this sounds good!

Reply:
Done